libyul/optimiser/Suite.cpp

```diff
@@ -140,6 +141,8 @@ void OptimiserSuite::run(
 		if (ast.statements.size() > 1 && std::get<Block>(ast.statements.front()).statements.empty())
 			ast.statements.erase(ast.statements.begin());
 	}
+
```

Comment:
Don't you think it makes sense to "reset" the name dispenser at this point?

Reply:
I added a reset.